tac/agents/v1/base/reactions.py

```diff
@@ -331,6 +334,19 @@ def _rejoin_tac(self, controller_pbk: str) -> None:
         msg = GetStateUpdate(self.crypto.public_key, self.crypto).serialize()
         self.out_box.out_queue.put(OutContainer(message=msg, message_id=0, dialogue_id=0, destination=controller_pbk))
 
+    def _search_for_tac(self) -> None:
+        """
+        Search for active TAC Controller.
+
+        We assume that the controller is registered as a service with the 'tac' data model
+        and with an attribute version = 1.
+
+        :return: None
+        """
+        query = Query([Constraint("version", GtEq(1))])
+        search_id = self.game_instance.search.get_next_id()
+        self.game_instance.search.ids_for_tac.add(search_id)
+        self.out_box.out_queue.put(OutContainer(query=query, search_id=search_id))
```

Comment:
This method is defined in `OEFReactions(OEFSearchReactionInterface)`. However, the same method is defined in `OEFActions(OEFSearchActionInterface)`. Unfortunately, it seems there is no way to call `OEFActions.search_for_tac` from a method of `OEFReactions` (more precisely, `OEFReactions._on_controller_search_result`).

Analogous issue for `ControllerActions.request_state_update` that needs to be called in `ControllerReactions.on_transaction_confirmed`:  https://github.com/fetchai/agents-tac/blob/v0.1.0/tac/agents/v1/base/reactions.py#L118

An idea might be to revise this bit of the design. Some thoughts:
- If both the reactions and actions implementations were in the same class, an action method could call a reaction method, and vice-versa. Hence, the `OEFHandler`, instead of inheriting the implementations from `OEFActions` and `OEFReactions`, it could inherit from `OEFSearchActionInterface` and `OEFSearchReactionInterface` and provide the implementations of the interfaces.
- The `OEFReactions` and `OEFActions` objects might have a reference to the `OEFHandler`. So for example, in the body of `OEFReactions._on_controller_search_result`, one could write `self.oef_handler.actions.search_for_tac()`.

Reply:
I agree on the issue of duplication and that being unacceptable. Your suggestion on design revision is interesting, though it would lead to a very bloated xHandler objects and less of a logical separation. However, at least in the case of the current PR I propose a different solution which does not create this issue:
- the `search_for_tac` is definitely an action and not a reaction. We already have sleep the agent loop and this additional sleep is messy. I would rather use the `is_competing` variable to keep track of whether the tac is found or not which can be updated upon successful/unsuccessful tac search result. Rather than set `self._is_competing = True` after search is initiated (line 104, participant_agent.py), we keep it False as long as the tac search is unsuccessful and so the agent will keep searching on each act without further changes required.

Reply:
Agreed, thank you! I'll switch to your solution.